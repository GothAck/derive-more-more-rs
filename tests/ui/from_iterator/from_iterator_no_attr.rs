use derive_more_more::FromIterator;

#[derive(Default, FromIterator)]
struct NamedMulti {
    inner: Vec<usize>,
    test: bool,
}

#[derive(Default, FromIterator)]
struct NamedSingle {
    inner: Vec<usize>,
}

#[derive(Default, FromIterator)]
struct UnnamedMulti(Vec<usize>, bool);

#[derive(Default, FromIterator)]
struct UnnamedSingle(Vec<usize>);

#[derive(Default, FromIterator)]
struct Unit;

fn main() {}
