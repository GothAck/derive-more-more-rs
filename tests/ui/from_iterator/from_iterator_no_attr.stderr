error: Expected `from_iterator` attribute on struct or one field
 --> tests/ui/from_iterator/from_iterator_no_attr.rs:4:1
  |
4 | struct NamedMulti {
  | ^^^^^^

error: Expected `from_iterator` attribute on struct or one field
  --> tests/ui/from_iterator/from_iterator_no_attr.rs:10:1
   |
10 | struct NamedSingle {
   | ^^^^^^

error: Expected `from_iterator` attribute on struct or one field
  --> tests/ui/from_iterator/from_iterator_no_attr.rs:15:1
   |
15 | struct UnnamedMulti(Vec<usize>, bool);
   | ^^^^^^

error: Expected `from_iterator` attribute on struct or one field
  --> tests/ui/from_iterator/from_iterator_no_attr.rs:18:1
   |
18 | struct UnnamedSingle(Vec<usize>);
   | ^^^^^^

error: Struct has no fields
  --> tests/ui/from_iterator/from_iterator_no_attr.rs:21:1
   |
21 | struct Unit;
   | ^^^^^^
